### PR TITLE
Fix warp events

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Fixes the event packing for data for `SendWarpMessage`.

Closes: #894

## How this works

Uses `abi` package `PackEvent`/`UnpackIntoInterface` functions to properly pack/unpack messages for event data.

## How this was tested

Adds e2e and UT tests.

### CODE REVIEW

File: scripts/run_ginkgo.sh [hunk 1]
```diff
@@ -37,3 +37,7 @@ TEST_SOURCE_ROOT="$TEST_SOURCE_ROOT" ginkgo run -procs=5 tests/precompile \
 ./tests/warp/warp.test \
   --ginkgo.vv \
   --ginkgo.label-filter=${GINKGO_LABEL_FILTER:-""}
+
+ ./tests/precompile/warp/warp.test \
```
Comment (darioush) [inline scripts/run_ginkgo.sh:41]: not sure we need a new directory for these tests? could we add them to the existing warp test instead? or extend the existing warp test to verify the events match the abi?

File: x/warp/contract.go [hunk 2]
```diff
@@ -256,22 +260,41 @@ func sendWarpMessage(accessibleState contract.AccessibleState, caller common.Add
...
 		accessibleState.GetBlockContext().Number().Uint64(),
 	)
 
 	// Return an empty output and the remaining gas
 	return []byte{}, remainingGas, nil
 }
 
+// PackSendWarpMessageEvent packs the given arguments into SendWarpMessage events including topics and data.
+func PackSendWarpMessageEvent(destinationChainID common.Hash, destinationAddress common.Address, sourceAddress common.Address, unsignedMessageBytes []byte) ([]common.Hash, []byte, error) {
+	return WarpABI.PackEvent("SendWarpMessage", destinationChainID, destinationAddress, sourceAddress, unsignedMessageBytes)
```
Comment (darioush) [inline x/warp/contract.go:285]: Seems we will be removing dstchain/dstaddr, so we should rebase this PR after the other work is merged

Comment (ceyonur) [reply x/warp/contract.go:285]: yea, we will add new topic as well.

File: scripts/run_ginkgo.sh [hunk 3]
```diff
@@ -37,3 +37,4 @@ TEST_SOURCE_ROOT="$TEST_SOURCE_ROOT" ginkgo run -procs=5 tests/precompile \
 ./tests/warp/warp.test \
   --ginkgo.vv \
   --ginkgo.label-filter=${GINKGO_LABEL_FILTER:-""}
+
```
Comment (darioush) [inline scripts/run_ginkgo.sh:40]: nit: let's just revert the whole file here for history?

File: tests/warp/warp_test.go [hunk 4]
```diff
@@ -380,4 +383,111 @@ var _ = ginkgo.Describe("[Warp]", ginkgo.Ordered, func() {
 		gomega.Expect(err).Should(gomega.BeNil())
 		gomega.Expect(receipt.Status).Should(gomega.Equal(types.ReceiptStatusSuccessful))
 	})
+
+	ginkgo.It("Send Message from A to B from hardhat", ginkgo.Label("Warp", "WarpContract", "SendWarpContract"), func() {
```
Comment (darioush) [inline tests/warp/warp_test.go:387]: probably could live in a separate file if possible?

Comment (ceyonur) [reply tests/warp/warp_test.go:387]: They share a lot of same variables from `var _ = ginkgo.Describe("[Warp]", ginkgo.Ordered, func() {`

**Review by aaronbuchwald — COMMENTED** [review]
Can we separate the warp hardhat test into a different PR and rebase with the `destinationChainID` and `destinationAddress` fields removed?

**Comment by ceyonur** [comment]
Closing in favor of #935 #937

**Pull request closed without merging**